test/bc_coeff_compositions.jl

```diff
@@ -433,25 +445,57 @@ end
     analytic_Qb = [zeros(N+1); sin(0.2+dx)]
     analytic_Ab = analytic_L*analytic_Qb
 
-    analytic_u = analytic_AL \ (u - analytic_Ab)
+
+    analytic_QM = zeros((length(x)+2)*3, length(x)*3)
+    interior = CartesianIndices(Tuple([2:length(x)+1, 1:3]))
+    I1 = CartesianIndex(1,0)
+    for I in interior
+        i = c2l(I, (length(x)+2, 3)) #helper function, see utils.jl
+        j = c2l(I-I1, (length(x), 3))
+        analytic_QM[i,j] = 1.0
+    end
+    analytic_Am = kron(Diagonal(ones(3)), analytic_L)*analytic_QM
+
+    analytic_u = analytic_AL \ Vector(u .- analytic_Ab)
     ghost_u = A \ u
 
-    # Check that A\u.(x) is consistent with analytic_AL \ u.(x)
-    @test analytic_u ≈ ghost_u
+    As_l, As_b = sparse(A, length(u))
 
-    # Check ldiv!
-    u_temp = zeros(N)
-    ldiv!(u_temp, A, u)
-    @test u_temp ≈ ghost_u ≈ analytic_u
+    @test As_l ≈ analytic_AL #This is true
+    @test As_b ≈  analytic_Ab #This is true
+    @test Vector(u .- analytic_Ab) ≈ Vector(u .- As_b) #this is true
+    @test_broken analytic_AL\Vector(u .- analytic_Ab) ≈ As_l\Vector(u .- As_b) #Then why isn't this working?

```

Comment:
@ChrisRackauckas  this error and those just below are what stalled progress before - I really have no idea why this isn't working

Reply:
what's the error?

Reply:
the test marked broken should definitely pass if the above three are passing

Reply:
It feels like it has to be a bug in the one of the `\` implementations (or just imprecision, I guess). Running everything in the REPL:

```
julia> using DiffEqOperators

julia> dx = 0.01; x = 0.01:dx:0.2; N=length(x); u=sin.(x); L = CenteredDifference(4, 4, dx, N)
; Q = RobinBC((1.0, 0.0, 0.0), (1.0, 0.0, sin(0.2+dx)), dx); A = L*Q;                         
                                                                                              
julia> analytic_L = fourth_deriv_approx_stencil(N) ./ dx^4; analytic_QL = [transpose(zeros(N))
; Diagonal(ones(N)); transpose(zeros(N))]; analytic_AL = analytic_L * analytic_QL; analytic_Qb
 = [zeros(N+1); sin(0.2+dx)]; analytic_Ab = analytic_L*analytic_Qb; 

julia> analytic_AL \ Vector(u .- analytic_Ab) ≈ As_l \ Vector(u .- As_b)
false

julia> analytic_AL \ Vector(u .- analytic_Ab) ≈ Array(As_l) \ Vector(u .- As_b)
true

julia> methods(\, typeof.([analytic_AL, Vector(u .- analytic_Ab)]))
# 1 method for generic function "\":                                                                                         
[1] \(A::AbstractArray{T,2} where T, B::Union{AbstractArray{T,1}, AbstractArray{T,2}} where T)
 in LinearAlgebra at /home/graham/.local/opt/julia-1.5.0/share/julia/stdlib/v1.5/LinearAlgebra
/src/generic.jl:1102

julia> methods(\, typeof.([As_l, Vector(u .- As_b)]))
# 1 method for generic function "\":
[1] \(A::SparseArrays.AbstractSparseMatrixCSC, B::Union{AbstractArray{T,1}, AbstractArray{T,2}} where T) in SparseArrays at /home/graham/.local/opt/julia-1.5.0/share/julia/stdlib/v1.5/SparseArrays/src/linalg.jl:1460  
```

Reply:
I filed an issue with an MWE: https://github.com/JuliaLang/julia/issues/37004.  I'll mark this as broken for now and move on.

Reply:
They pointed out that the difference is within the expected range given the condition number of `analytic_AL`. I've revised the test to reflect that, and now it and a nearby similar test are passing. I imagine this is a problem in some other tests as well.  Would we rather use a better conditioned matrix?  I wouldn't know how to choose.

Reply:
Just change the tolerance to be fit for the condition number. Any stencil operation is going to be pretty ill-conditioned (this is where the stiffness comes from)